### DISCUSSION TITLE
How to make 2D images look the best quality

### DISCUSSION THREAD

**Discussion by rdarnold** [MonoGame/MonoGame#8595] [Graphics]
I have a game where I draw high-res sprites scaled down on screen.  I am having trouble getting them to look the equivalent of how they look in MS Photos when scaled to the same size.  I've tried:

- Changing SpriteBatch SamplerState to all the different states.  The default seems to look the best, but nothing looks great.
- Generating Mipmaps for the images in MGCB.  This results in the images anti-aliasing better, but also looking blurry.

Anything else I can do?  They look great when I load them up in MS Photos and scale them to the same size that I try to scale them in my game.  So I know they can draw in a way that looks good without me editing the images by hand to change their native res (I have hundreds).  I just don't know how to make it happen in Monogame.

**Comment by KelsamGames** [comment]
Hmm. Well there are a lot of potential variables at play here. Are you drawing to an RT that is later resized and drawn to the backbuffer? If so, have you enabled mipmapping there as well? Have you drawn and compared your full-rez images side-by-side to rule out anything happening with the MGCB compiling process doing anything weird to the images (and/or tried different compile/compression fomats)? If you've ruled out all "this could be derping this up" things like the examples above, and it still looks significantly different, then I think your only option would be to write (or find) a custom scaling shader. You could look into scaling algorithms like "nearest neighbor", "bicubic resampling", etc to see what sort of end-result you're looking for.

**Reply by KelsamGames** [reply]
Another thought, have you played around with default multisampling stuff at all?

Ex:
```
_graphics.GraphicsProfile = GraphicsProfile.HiDef;
_graphics.PreferMultiSampling = true;
_graphics.GraphicsDevice.PresentationParameters.MultiSampleCount = 8;
_graphics.ApplyChanges();
```

**Comment by rdarnold** [comment]
I just tried, at random, creating my own SamplerState and passing that to the SpriteBatch begin method:

```
SamplerState ss = new SamplerState();
ss.Filter = TextureFilter.Anisotropic;
ss.MipMapLevelOfDetailBias = -1;
```

If I use this method and select "generate mipmaps" through MGCB, the results are really good.  I don't understand what MipMapLevelOfDetailBias is doing.  I just set it to -1 at random.
